apps/server/src/modules/common-cartridge/common-cartridge-client/lesson-client/dto/component-lernstore-props.dto.ts

```diff
@@ -1,9 +1,10 @@
 import { ComponentLernstorePropsImpl } from '../lessons-api-client';
+import { LernstoreResourcesDto } from './lernstore-resources.dto';
 
 export class ComponentLernstorePropsDto {
-	public resources: string[];
+	public resources: LernstoreResourcesDto[];
 
 	constructor(lernstoreContent: ComponentLernstorePropsImpl) {
-		this.resources = lernstoreContent.resources;
+		this.resources = lernstoreContent.resources.map((resource) => new LernstoreResourcesDto(resource));
 	}
 }
```

Comment:
Couldn't we do it like this and put the rest in the new method in lesson-dto.mapper.ts?

```suggestion
export class ComponentLernstorePropsDto {
    public resources: LernstoreResourcesDto[];

    constructor(resources: LernstoreResourcesDto[]) {
        this.resources = resources;
    }
}
```